components/calendar/src/astronomy.rs

```diff
@@ -226,6 +258,140 @@ impl Astronomical {
+                + 4.0 * eccentricity * y * sin_degrees(anomaly) * cos_degrees(2.0 * lambda)
+                - 0.5 * libm::pow(y, 2.0) * sin_degrees(4.0 * lambda)
+                - 1.25 * libm::pow(eccentricity, 2.0) * sin_degrees(2.0 * anomaly));
+
+        signum(equation) * libm::fabs(equation).min(12.0 / 24.0)
+    }
+
+    #[allow(dead_code, clippy::unwrap_used)]
+    pub(crate) fn dusk(date: f64, location: Location, alpha: f64) -> Option<Moment> {
+        let evening = false;
+        let result =
+            Self::moment_of_depression(Moment::new(date + (18.0 / 24.0)), location, alpha, evening);
+
+        result?;
+        Some(Location::standard_from_local(result.unwrap(), location))
+    }
+
+    // Calculates the obliquity of the ecliptic at a given moment
+    pub fn obliquity(moment: Moment) -> f64 {
+        let c = Self::julian_centuries(moment);
+        let angle = 23.0 + 26.0 / 60.0 + 21.448 / 3600.0;
+        let list = [0.0, -46.8150 / 3600.0, -0.00059 / 3600.0, 0.001813 / 3600.0];
+
+        let mut result = angle;
+        let mut c_power = c;
+        for coef in list.into_iter().skip(1) {
+            result += coef * c_power;
+            c_power *= c;
+        }
+        result
+    }
+    // Calculates declination at a given Moment of UTC time for the latitude and longitude of an object lambda
+    pub(crate) fn declination(moment: Moment, beta: f64, lambda: f64) -> f64 {
+        let varepsilon = Self::obliquity(moment);
+        arcsin_degrees(
+            sin_degrees(beta) * cos_degrees(varepsilon)
+                + cos_degrees(beta) * sin_degrees(varepsilon) * sin_degrees(lambda),
+        )
+    }
+    pub(crate) fn right_ascension(
+        moment: Moment,
+        beta: f64,
+        lambda: f64,
+    ) -> Result<f64, &'static str> {
+        let varepsilon = Self::obliquity(moment);
+        arctan_degrees(
+            sin_degrees(lambda) * cos_degrees(varepsilon)
+                - tan_degrees(beta) * sin_degrees(varepsilon),
+            cos_degrees(lambda),
+        )
+    }
+
+    pub(crate) fn local_from_apparent(moment: Moment, location: Location) -> Moment {
+        moment - Self::equation_of_time(Location::universal_from_local(moment, location))
+    }
+
+    pub(crate) fn approx_moment_of_depression(
+        moment: Moment,
+        location: Location,
+        alpha: f64,
+        early: bool,
+    ) -> Option<Moment> {
+        let mut t = Self::sine_offset(moment, location, alpha);
+        let date = moment.as_rata_die().to_i64_date() as f64;
+        let alt = if alpha >= 0.0 {
+            if early {
+                date
+            } else {
+                date + 1.0
+            }
+        } else {
+            date + 12.0 / 24.0
+        };
+
+        let value = if libm::fabs(t) > 1.0 {
+            t = Self::sine_offset(Moment::new(alt), location, alpha);
+            t
+        } else {
+            t
```

Comment:
If you don't modify `t` in the "then branch" of the if-then-else block, then you can avoid declaring `t` locally, and replace this line with the expression you used to initialize `t`.